extensions/ollama/src/ollama-extension.ts

```diff
@@ -0,0 +1,115 @@
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import { type Disposable, type ExtensionContext, type Provider, provider } from '@kortex-app/api';
+import { createOllama } from 'ollama-ai-provider-v2';
+
+export class OllamaExtension {
+  #extensionContext: ExtensionContext;
+  #currentModels: string[] = [];
+  #connectionDisposable: Disposable | undefined;
+  #interval: NodeJS.Timeout | undefined;
+
+  constructor(extensionContext: ExtensionContext) {
+    this.#extensionContext = extensionContext;
+  }
+
+  async activate(): Promise<void> {
+    const ollamaProvider = provider.createProvider({
+      name: 'Ollama',
+      status: 'unknown',
+      id: 'ollama',
+      images: {
+        icon: './icon.png',
+        logo: {
+          dark: './icon.png',
+          light: './icon.png',
+        },
+      },
+    });
+    this.#extensionContext.subscriptions.push(ollamaProvider);
+
+    await this.updateModelsAndStatus(ollamaProvider);
+    this.#interval = setInterval(() => {
+      this.updateModelsAndStatus(ollamaProvider).catch((error: unknown) => {
+        console.error('Error updating Ollama models and status:', error);
+      });
+    }, 30000);
+  }
+
+  protected async updateModelsAndStatus(ollamaProvider: Provider): Promise<void> {
+    let models: Array<{ name: string }> = [];
+    let running = true;
+    try {
+      const res = await fetch('http://localhost:11434/api/tags');
+      if (!res.ok) throw new Error('Ollama not running');
+      const data = await res.json();
+      models = Array.isArray(data.models) ? data.models : [];
+    } catch (_err: unknown) {
+      running = false;
+      models = [];
+    }
+
+    // Update provider status
+    if (!running) {
+      ollamaProvider.updateStatus('stopped');
+      // deregister previous connection if exists
+      if (this.#connectionDisposable) {
+        this.#connectionDisposable.dispose();
+        this.#connectionDisposable = undefined;
+      }
+      this.#currentModels = [];
+      return;
+    }
+
+    ollamaProvider.updateStatus('started');
+    const newModelNames = models.map(m => m.name).sort((a, b) => a.localeCompare(b));
+    const oldModelNames = this.#currentModels.slice().sort((a, b) => a.localeCompare(b));
+    const modelsChanged =
+      newModelNames.length !== oldModelNames.length || newModelNames.some((v, i) => v !== oldModelNames[i]);
+
+    if (modelsChanged) {
+      // Unregister previous connection if exists
+      if (this.#connectionDisposable) {
+        this.#connectionDisposable.dispose();
+        this.#connectionDisposable = undefined;
+      }
+      this.#currentModels = newModelNames;
+      if (newModelNames.length > 0) {
+        const sdk = createOllama();
+        const disposable = ollamaProvider.registerInferenceProviderConnection({
+          name: 'ollama',
+          sdk,
+          status() {
+            return running ? 'started' : 'stopped';
```

Comment:
<img width="499" height="178" alt="Image" src="https://github.com/user-attachments/assets/889d5282-c71a-498c-97ca-d3b224d1024c" />

It is possible to return 'started' always here as running is true (see screenshot)